pkgs/racket-test/tests/racket/contract/test-util.rkt

```diff
@@ -360,6 +361,20 @@
         [(string? blame) (string-append "blaming: " (regexp-quote blame))]
         [else #f]))
     (and reg (regexp-match? reg msg)))
+  (define (has-proper-important-blame? msg)
+    (cond
+      [(not important)]
+      [(not (or (eq? blame 'pos) (eq? blame 'neg)))
+       #f]
+      [else
+       (define important-name (car important))
+       (define important-party (cdr important))
+       (define self-or-not
+         (if (equal? (eq? blame 'pos) (eq? important-party 'pos))
+             "broke its own contract"
+             "contract violation"))
+       (define reg (string-append (regexp-quote important-name) ": " self-or-not))
+       (regexp-match? reg msg)]))
   (contract-eval
```

Comment:
actually, how about a different strategy here. I think there are two things this is checking: if "broke its own" or "contract violation" occurs, and then what name appears. Can we make that be two parameters to the function, with a default value that means "don't check this part of the test"?